.clang-tidy

```diff
@@ -64,7 +64,9 @@ Checks: '*,
          -readability-named-parameter,
          -misc-no-recursion,
          -concurrency-mt-unsafe,
-         -bugprone-easily-swappable-parameters'
+         -bugprone-easily-swappable-parameters,
+         -cppcoreguidelines-macro-usage,
```

Comment:
I have added those two, because `//NOLINTNEXTLINE`is severely broken for macros and `//NOLINTBEGIN/END` are available on clang-tidy version 14 (we are on 13). The solution, (which I wanted to opt in while I was migrating stuff) is to introduce a template. However, I am not keen into making a lot of changes while I don't have all the unit tests I would like. Once we move forward, the macros will be replaced with templates and these clang-tidy warnings shall be removed.

Reply:
I am not sure why but I think because the macro is multiline. I have tried `//NOLINTNEXTLINE(bugprone-easily-swappable-parameters, cppcoreguidelines-macro-usage)` but it did not work. If you know why or how to make it work I am more than happy to apply it :)